src/adu-shell/src/main.cpp

```diff
@@ -265,16 +277,16 @@ bool ADUShell_PermissionCheck()
     bool isTrusted = false;
 
     // If config file is provided, check if user is in trusted user list.
-    ADUC_ConfigInfo config = {};
-    if (ADUC_ConfigInfo_Init(&config, ADUC_CONF_FILE_PATH))
+    const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
+    if (config != NULL)
```

Comment:
if config == NULL, then caller needs to return ENOMEM rather than EPERM. Maybe can just remove NULL check. When would it happen?

Reply:
The config can be null if du-config.json file is missing or not accessible, or any required field is missing.